accounting_reporting_core/src/test/java/org/cardanofoundation/lob/app/accounting_reporting_core/repository/PublicReportRepositoryTest.java

```diff
@@ -0,0 +1,95 @@
+                LEFT JOIN accounting_reporting_core.report.ReportEntity r2 on r.idControl = r2.idControl and r.ver < r2.ver
+                WHERE r2.idControl IS NULL
+                AND r.ledgerDispatchStatus = 'FINALIZED'
+                AND r.type = 'INCOME_STATEMENT'
+                AND r.intervalType = 'MONTH'
+                AND r.year = 2025
+                AND r.period = 1
+                ORDER BY r.createdAt ASC, r.reportId ASC
+                """;
+        jakarta.persistence.Query queryResult = Mockito.mock(Query.class);
+        PublicReportRepository publicReportRepository = new PublicReportRepository(em);
+        Mockito.when(em.createQuery(Mockito.anyString())).thenReturn(queryResult);
+        publicReportRepository.findAllByTypeAndPeriod(ReportType.INCOME_STATEMENT, IntervalType.MONTH, (short) 2025, (short) 1);
+        Mockito.verify(em, Mockito.times(1)).createQuery(query);
+    }
+
+    @Test
+    void findAllByTypeAndPeriodTestNoReportType() {
+        String query = """
+                SELECT r FROM accounting_reporting_core.report.ReportEntity r
+                LEFT JOIN accounting_reporting_core.report.ReportEntity r2 on r.idControl = r2.idControl and r.ver < r2.ver
+                WHERE r2.idControl IS NULL
+                AND r.ledgerDispatchStatus = 'FINALIZED'
+                AND r.intervalType = 'MONTH'
+                AND r.year = 2025
+                AND r.period = 1
+                ORDER BY r.createdAt ASC, r.reportId ASC
+                """;
+        jakarta.persistence.Query queryResult = Mockito.mock(Query.class);
+        PublicReportRepository publicReportRepository = new PublicReportRepository(em);
+        Mockito.when(em.createQuery(Mockito.anyString())).thenReturn(queryResult);
+        publicReportRepository.findAllByTypeAndPeriod(null, IntervalType.MONTH, (short) 2025, (short) 1);
+        Mockito.verify(em, Mockito.times(1)).createQuery(query);
+    }
+
+    @Test
+    void findAllByTypeAndPeriodTestNoIntervalType() {
+        String query = """
+                SELECT r FROM accounting_reporting_core.report.ReportEntity r
+                LEFT JOIN accounting_reporting_core.report.ReportEntity r2 on r.idControl = r2.idControl and r.ver < r2.ver
+                WHERE r2.idControl IS NULL
+                AND r.ledgerDispatchStatus = 'FINALIZED'
+                AND r.type = 'BALANCE_SHEET'
+                ORDER BY r.createdAt ASC, r.reportId ASC
+                """;
+        jakarta.persistence.Query queryResult = Mockito.mock(Query.class);
+        PublicReportRepository publicReportRepository = new PublicReportRepository(em);
+        Mockito.when(em.createQuery(Mockito.anyString())).thenReturn(queryResult);
+        publicReportRepository.findAllByTypeAndPeriod(ReportType.BALANCE_SHEET,null, (short) 2025, (short) 1);
+        Mockito.verify(em, Mockito.times(1)).createQuery(query);
+    }
+
+    @Test
+    void findAllByTypeAndPeriodTestNofilter() {
```

Comment:
Mutation tests aside, it would be good if the query actually works but maybe this is more like functional spring boot test.